llvmcpy/llvm.py

```diff
@@ -257,6 +258,28 @@ def call():
         if failure != 0:
             raise LLVMException({})
         return result""".format(result_type, call(), error_message)
+    elif (return_type.kind == "pointer"
+          and return_type.item.kind == "primitive"
+          and return_type.item.cname == "char"
+          and last_arg and last_arg.kind == "pointer"
+          and last_arg.item.kind == "primitive"
+          and last_arg.item.cname in ("unsigned int", "unsigned long")):
+        # Special case: the function returns a string and
+        # the last argument is a pointer to an unsigned int/long:
+        # This often represents returning a string (that may contain
+        # NUL characters) with a length
+        function_arguments[-1] = None
+        arguments[-1] = "lengthbuf"
+        # Print function header
+        result += header()
+        # Print the function body: first create a buffer for the length, then
+        # call the function and build the string return value from the
+        # returned char pointer and the length
+        result += """        lengthbuf = ffi.new("{}[1]")
+        ptr = {}
+        length = lengthbuf[0]
+        return ffi.unpack(ptr, length)""".format(last_arg.item.cname, call())
```

Comment:
Right, we should check elsewhere if there's some erroneous usage of `ffi.string`, but I suppose they're fine.